src/redux/footballSlice.js

```diff
@@ -0,0 +1,34 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+  loading: false,
+  error: null,
+  teams: [],
+};
+
+export const fetchTeams = createAsyncThunk('teams/fetchTeams', async () => {
+  const resp = await axios('https://www.thesportsdb.com/api/v1/json/3/search_all_teams.php?l=English%20Premier%20League');
+  return resp.data.teams;
+});
+
+const teamsSlice = createSlice({
+  name: 'teams',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchTeams.pending, (state) => {
+      state.loading = true;
+    });
+    builder.addCase(fetchTeams.fulfilled, (state, action) => {
+      state.loading = false;
+      state.teams = action.payload;
+    });
+    builder.addCase(fetchTeams.rejected, (state, action) => {
+      state.loading = false;
+      state.error = action.error.message;
+    });
+  },
+});
```

Comment:
- [ ]  Hello! 🎓 I'd like to emphasize the importance of avoiding direct state changes, even when using createSlice with Immer under the hood. While Immer simplifies state mutations, it's still a good practice to work with Redux state immutably.
Instead of directly modifying state properties, consider using spreading or concatenation to create new state objects. This helps maintain code clarity, predictability, and ensures that you're following best practices for Redux.

For instance, when updating an array in your state, use methods like concat or the spread operator (...) to create a new array with the changes. When updating an object, create a new object with the spread operator. This approach not only aligns with Redux principles but also makes your code more readable and easier to debug.

Remember, one of the core principles of Redux is immutability, and it's a good habit to follow even when tools like createSlice abstract some of the complexity. It'll lead to cleaner, more maintainable code in the long run. 😊🧹

So, keep up the great work, and remember that coding practices that promote immutability contribute to more robust and maintainable applications! 🚀📚